viam-cartographer/scripts/build_viam_cartographer_debug.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+set -o errexit
+
+if [ "$(uname)" == "Linux" ]; then
+    cd $(dirname $(readlink -f "${BASH_SOURCE}"))/..
+elif [ "$(uname)" == "Darwin" ]; then
+    readlinkorreal() { readlink "$1" || echo "$1"; }
+    cd $(dirname string readlinkorreal "${BASH_SOURCE}")/..
+    which brew && export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:`brew --prefix openssl@3`/lib/pkgconfig"
+else
+    echo ERROR: your OS is not handled yet
+    exit 1
+fi
+
+# Build and install Cartographer with the viam wrapper.
+DIR=`pwd`/build/
+echo $DIR
+if [ ! -d "$DIR" ]; then
+    mkdir build
+else 
+    echo "Build Directory already exists"
+fi
+
+pushd build
+
+echo building viam-cartographer
+cmake .. -G Ninja \
+  -DCMAKE_CXX_STANDARD=17 \
+  -DCMAKE_PREFIX_PATH=`brew --prefix` \
+  -DQt5_DIR=$(brew --prefix qt5)/lib/cmake/Qt5 \
+  -DCMAKE_BUILD_TYPE=Debug
```

Comment:
b/c the normal build doesn't enable debug we can't run cartographer through GDB.  This script is the same as build_viam_cartographer.sh + this line. In the future this also gives us a place to enable ASAN support: https://viam.atlassian.net/browse/RSDK-3538